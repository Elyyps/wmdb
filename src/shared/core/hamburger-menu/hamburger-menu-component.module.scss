.hamburger-menu {
  height: 100%;
  overflow: auto;
  &__navigation {
    li {
      margin-bottom: 8px;
      a {
        text-decoration: none;
        color: $global-inverse-color;
        &:hover {
          color: $global-link-color;
        }
      }
    }
  }
}
