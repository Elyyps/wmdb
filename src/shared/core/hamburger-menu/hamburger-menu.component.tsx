import * as React from "react";
import styles from "./hamburger-menu-component.module.scss";
import { LinkComponent } from "../link";
import { ILink } from "@app/api/core/link/link";

export interface IHamburgerMenuComponentProps {
  menuItems: ILink[];
}

const HamburgerMenuComponent = (props: IHamburgerMenuComponentProps) => (
  <div className={styles["hamburger-menu"]}>
    <ul className={styles["hamburger-menu__navigation"]}>
      {props.menuItems.map((item, key) => (
        <li key={key}>
          <LinkComponent to={item.url}>{item.title}</LinkComponent>
        </li>
      ))}
    </ul>
  </div>
);

export { HamburgerMenuComponent };
