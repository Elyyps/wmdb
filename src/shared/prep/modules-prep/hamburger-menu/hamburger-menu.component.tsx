import * as React from "react";
import "./hamburger-menu.component.scss";
import { Link } from "react-router-dom";

export interface IHamburgerMenuComponentProps {}

const HamburgerMenuComponent = (props: IHamburgerMenuComponentProps) => (
  <div className="hamburger-menu">
    <ul className="hamburger-menu__navigation">
      <li>
        <Link to="./">Lorem</Link>
      </li>
      <li>
        <Link to="./">Lorem</Link>
      </li>
      <li>
        <Link to="./">Lorem</Link>
      </li>
    </ul>
  </div>
);

export { HamburgerMenuComponent };
