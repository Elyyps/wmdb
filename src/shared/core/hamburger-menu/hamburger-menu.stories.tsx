import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { HamburgerMenuComponent } from "./hamburger-menu.component";

storiesOf("HamburgerMenu", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <HamburgerMenuComponent menuItems={[]} />);
