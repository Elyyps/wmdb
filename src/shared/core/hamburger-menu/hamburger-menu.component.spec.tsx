import React from "react";
import { shallow } from "enzyme";
import { HamburgerMenuComponent } from "./hamburger-menu.component";

describe("[HamburgerMenu]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HamburgerMenuComponent menuItems={[]} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
