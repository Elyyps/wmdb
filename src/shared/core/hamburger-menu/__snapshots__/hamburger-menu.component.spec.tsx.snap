// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HamburgerMenu] should render component without crashing 1`] = `
<div
  className="hamburger-menu"
>
  <ul
    className="hamburger-menu__navigation"
  />
</div>
`;
